jobs/bosh-dns/spec

```diff
@@ -122,6 +122,10 @@ properties:
     default:
       - upcheck.bosh-dns.
 
+  internal_check_domain:
+    description: "Domain name to use to validate internal resolution is working"
+    default: "cloud-controller-ng.service.cf.internal"
```

Comment:
I think I would prefer having each VM resolve itself (either leveraging https://bosh.io/docs/jobs/#properties-spec in the templates, or perhaps using a reverse lookup on the local IP to get the FQDN then resolving that.

bosh-dns is deployment agnostic, so a CF alias isn't an appropriate default. And if VMs are resolving another internal VM as an upcheck, that means two things:
- they'll all be polling the health server on that VM (extra load)
- when that one VM is unhealthy (e.g. being deployed) then bosh-dns will restart on every other VM, and depending on timing that could cause the deployment to fail